jobs/async-upload/tests/test_models.py

```diff
@@ -0,0 +1,273 @@
+
+    def test_create_model_intent_no_validation(self):
+        """Test that create_model intent has no validation requirements"""
+        # Should succeed even with empty strings since create_model has no validation
+        config = ModelConfig(
+            upload_intent=UploadIntent.create_model,
+            id="",  # Empty string - should be fine
+            version_id="",  # Empty string - should be fine
+            artifact_id=""  # Empty string - should be fine
+        )
+        
+        # Should not raise any validation errors
+        assert config.upload_intent == UploadIntent.create_model
+        assert config.id == ""
+        assert config.version_id == ""
+        assert config.artifact_id == ""
+
+    def test_create_model_intent_with_partial_data_succeeds(self):
+        """Test that create_model intent succeeds with partial data"""
+        config = ModelConfig(
+            upload_intent=UploadIntent.create_model,
+            id="test-model-id",
+            version_id="",  # Empty string - should be fine
+            artifact_id="test-artifact-id"
+        )
+        
+        # Should not raise any validation errors
+        assert config.upload_intent == UploadIntent.create_model
+        assert config.id == "test-model-id"
+        assert config.version_id == ""
+        assert config.artifact_id == "test-artifact-id"
+
+    def test_create_model_intent_with_mixed_data_succeeds(self):
+        """Test that create_model intent succeeds with mixed data"""
+        config = ModelConfig(
+            upload_intent=UploadIntent.create_model,
+            id="test-model-id",
+            version_id="test-version-id",
+            artifact_id=""  # Empty string - should be fine
+        )
+        
+        # Should not raise any validation errors
+        assert config.upload_intent == UploadIntent.create_model
+        assert config.id == "test-model-id"
+        assert config.version_id == "test-version-id"
+        assert config.artifact_id == ""
+
+    def test_create_model_intent_allows_all_empty_ids(self):
+        """Test that create_model intent allows all empty IDs"""
+        config = ModelConfig(
+            upload_intent=UploadIntent.create_model,
+            id="",
+            version_id="",
+            artifact_id=""
+        )
+        
+        # Should not raise any validation errors
+        assert config.upload_intent == UploadIntent.create_model
+        assert config.id == ""
+        assert config.version_id == ""
+        assert config.artifact_id == ""
+
+    def test_create_version_intent_with_required_ids_succeeds(self):
+        """Test that create_version intent succeeds when model ID and version ID are provided"""
+        config = ModelConfig(
+            upload_intent=UploadIntent.create_version,
+            id="test-model-id",
+            version_id="test-version-id",
+            artifact_id="test-artifact-id"  # Can be provided but not required
```

Comment:
analogous question to above, not sure I got the intent of this test; if this is meant to create the tuple for ModelVersion and ModelArtifact, why some IDs for all entities are supplied?